modules/dreamview/backend/simulation_world/simulation_world_service.cc

```diff
@@ -1013,6 +1013,20 @@ void SimulationWorldService::UpdateSimulationWorld(
 template <>
 void SimulationWorldService::UpdateSimulationWorld(
     const NavigationInfo &navigation_info) {
+  static const std::string topic_name = "NavigationInfo";
+  if (navigation_info.has_hashcode()) {
+    const auto hashcode = navigation_info.hashcode();
+    if (topic_hashcode_map_.find(topic_name) !=
+        topic_hashcode_map_.end()) {
+      // it seems to be the same topic, ignore this update
+      if (topic_hashcode_map_[topic_name] == hashcode) {
+        return;
+      }
+    }
+
+    topic_hashcode_map_[topic_name] = hashcode;
```

Comment:
why do you need a map if the topic_name is always "NavigationInfo"?

Reply:
I considered that it's more convenience for further extension. What if another topic also needs to be distinguished via hashcode? So I used an unorderd_map to store hashcodes.

Reply:
I don't see a reason to optimize prematurely here, normally a module is not supposed to resend identical data. Also what you are really doing here is to dedup consecutive message with same content, just having a local static to store previous hash_code suffices.

Reply:
Thanks for your suggestion. Using timestamp to check and using local variable to store timestamp also meet the needs of this feature, simply and clean enough^^. I'll modify and commit today.